assets/js/modules/search-console/components/module/ModuleOverviewWidget.js

```diff
@@ -0,0 +1,326 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { Fragment, useState, useCallback } from '@wordpress/element';
+import { __, sprintf, _n, _x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_NAME, DATE_RANGE_OFFSET } from '../../datastore/constants';
+import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import { CORE_SITE } from '../../../../googlesitekit/datastore/site/constants';
+import { decodeHTMLEntity, getCurrentDateRangeDayCount } from '../../../../util';
+import {
+	extractSearchConsoleDashboardData,
+	getSiteStatsDataForGoogleChart,
+	generateDateRangeArgs,
+	isZeroReport,
+} from '../../util';
+import WidgetHeaderTitle from '../../../../googlesitekit/widgets/components/WidgetHeaderTitle';
+import WidgetHeaderCTA from '../../../../googlesitekit/widgets/components/WidgetHeaderCTA';
+import { Grid, Row, Cell } from '../../../../material-components';
+import PreviewBlock from '../../../../components/PreviewBlock';
+import GoogleChart from '../../../../components/GoogleChart';
+import DataBlock from '../../../../components/DataBlock';
+import Data from 'googlesitekit-data';
+const { useSelect } = Data;
+
+const ModuleOverviewWidget = ( { Widget, WidgetReportZero, WidgetReportError } ) => {
+	const [ selectedStats, setSelectedStats ] = useState( 0 );
+	const dateRange = useSelect( ( select ) => select( CORE_USER ).getDateRange() );
+	const { startDate, endDate, compareStartDate } = useSelect(
+		( select ) => select( CORE_USER ).getDateRangeDates(
+			{
+				compare: true,
+				offsetDays: DATE_RANGE_OFFSET,
+			}
+		)
+	);
+	const reportArgs = {
+		startDate: compareStartDate,
+		endDate,
+		dimensions: 'date',
+	};
+	const data = useSelect( ( select ) => select( STORE_NAME ).getReport( reportArgs ) );
+	const error = useSelect( ( select ) => select( STORE_NAME ).getErrorForSelector( 'getReport', [ reportArgs ] ) );
+	const loading = useSelect( ( select ) => ! select( STORE_NAME ).hasFinishedResolution( 'getReport', [ reportArgs ] ) );
+	const propertyID = useSelect( ( select ) => select( STORE_NAME ).getPropertyID() );
+	const currentEntityTitle = useSelect( ( select ) => select( CORE_SITE ).getCurrentEntityTitle() );
+	const searchConsoleDeepArgs = {
+		resource_id: propertyID,
+		...generateDateRangeArgs( { startDate, endDate } ),
+	};
+	const searchConsoleDeepLink = useSelect( ( select ) => select( STORE_NAME ).getServiceURL( {
+		path: '/performance/search-analytics',
+		query: searchConsoleDeepArgs,
+	} ) );
+
+	const handleStatSelection = useCallback( ( stat ) => {
+		setSelectedStats( stat );
+	}, [] );
+
+	if ( loading ) {
+		return <PreviewBlock width="100%" height="250px" />;
+	}
```

Comment:
@asvinb let's better display a progress bar to make it consistent with the behavior of the legacy component.

Reply:
👍